src/VoteHelper.ts

```diff
@@ -0,0 +1,161 @@
+        this.trackNumber = Number(trackNumber.toHuman())
+        this.data = trackVotes
+    }
+}
+
+export class VoteHelper {
+    private api: ApiPromise;
+
+    constructor(connection: SubstrateConnection) {
+        this.api = connection.getApi();
+    }
+
+    async checkVotes(network: NetworkConfig, sender): Promise<void> {
+        for (const validator of network.validators) {
+            const unvotedReferena = await this.getUnvotedReferenda();
+            const potentialVotes = await this.getPotentialVotes(network, unvotedReferena)
+            const newVotes = await this.decideHowToVote(validator, potentialVotes)
+
+            for (const vote of newVotes) {
+                await this.voteForReferenda(vote, sender, validator, network.voteAmount);
+            }
+        }
+    }
+
+    async getUnvotedReferenda(): Promise<Referendum[]> {
+        let referendums = await this.api.query.referenda.referendumInfoFor.entries()
+        const castedReferendums = referendums.map(([key, value]) => {
+            return new Referendum(key, value);
+        });
+
+        const ongoingReferendums = castedReferendums.filter(referendum => {
+            if (referendum.data.ongoing) {
+                return true;
+            } else {
+                return false;
+            }
+        })
+
+        return ongoingReferendums
+    }
+
+    async getAccountVotes(accountAddress: string): Promise<Vote[]> {
+        const accountVotes: Vote[] = []
+        const uniqueVotes = new Set<number>();
+
+        const referendaTracks = this.api.consts.referenda.tracks
+
+        const trackArray = (referendaTracks as any).map(([key, value]) => {
+            return new TrackVotes(key, value)
+        });
+
+        for (const track of trackArray) {
+            const votesInTrack = await this.api.query.convictionVoting.votingFor(accountAddress, track.trackNumber);
+            (votesInTrack as any).asCasting.votes.map((voteData) => {
+                const vote = new Vote(voteData)
+                if (!uniqueVotes.has(vote.referendaNumber)) {
+                    uniqueVotes.add(vote.referendaNumber)
+                    accountVotes.push(vote)
+                }
+            })
+        }
+
+        return accountVotes
+    }
+
+    async getPotentialVotes(network: NetworkConfig, referendums: Referendum[]): Promise<Vote[]> {
+        const uniqueVotes = new Set<number>();
+        const newVoteForReferenda: Vote[] = [];
+        for (const account of network.votersAccounts) {
+            const voterVotes = await this.getAccountVotes(account);
+
+            for (const referenda of referendums) {
+                const vote = voterVotes.find(voterVote => voterVote.referendaNumber === referenda.number);
+                if (vote) {
+                    if (!uniqueVotes.has(vote.referendaNumber)) {
+                        uniqueVotes.add(vote.referendaNumber)
+                        newVoteForReferenda.push(vote)
```

Comment:
this part is responsible for taking the first vote for the referendum, going through the list of voters from 0 to m, which provides a ranking in determining whose votes from the list we will follow